BlobStorage/Advanced.cs

```diff
@@ -1132,74 +934,68 @@ private static void PrintBlobPropertiesAndMetadata(CloudBlob blob)
+            }
         }
 
+
         /// <summary>
         /// Lists blobs in the specified container using a flat listing, with an optional segment size specified, and writes 
         /// their properties and metadata to the console window.
         /// The flat listing returns a segment containing all of the blobs matching the listing criteria.
         /// In a flat listing, blobs are not organized by virtual directory.
         /// </summary>
-        /// <param name="container">A CloudBlobContainer object.</param>
+        /// <param name="container">A BlobContainerClient object.</param>
         /// <param name="segmentSize">The size of the segment to return in each call to the listing operation.</param>
         /// <returns>A Task object.</returns>
-        private static async Task ListBlobsFlatListingAsync(CloudBlobContainer container, int? segmentSize)
+        private static async Task ListBlobsFlatListingAsync(BlobContainerClient container, int? segmentSize)
         {
             // List blobs to the console window.
             Console.WriteLine("List blobs in segments (flat listing):");
             Console.WriteLine();
 
-            int i = 0;
-            BlobContinuationToken continuationToken = null;
-            BlobResultSegment resultSegment = null;
-
             try
             {
                 // Call ListBlobsSegmentedAsync and enumerate the result segment returned, while the continuation token is non-null.
                 // When the continuation token is null, the last segment has been returned and execution can exit the loop.
-                do
+                // This overload allows control of the segment size. You can return all remaining results by passing null for the maxResults parameter, 
+                // or by calling a different overload.
+                // Note that requesting the blob's metadata as part of the listing operation 
+                // populates the metadata, so it's not necessary to call FetchAttributes() to read the metadata.
+                IAsyncEnumerable<Page<BlobItem>> pages = container.GetBlobsAsync(traits: BlobTraits.Metadata).AsPages(pageSizeHint: segmentSize);
+                await foreach (Page<BlobItem> page in pages)
                 {
-                    // This overload allows control of the segment size. You can return all remaining results by passing null for the maxResults parameter, 
-                    // or by calling a different overload.
-                    // Note that requesting the blob's metadata as part of the listing operation 
-                    // populates the metadata, so it's not necessary to call FetchAttributes() to read the metadata.
-                    resultSegment = await container.ListBlobsSegmentedAsync(string.Empty, true, BlobListingDetails.Metadata, segmentSize, continuationToken, null, null);
-                    if (resultSegment.Results.Count() > 0)
+                    foreach (BlobItem blobItem in page.Values)
```

Comment:
Are we intending to segment this loop? If the point is to demonstrate advanced controls then this is fine, but it looks like the segment size is an afterthought and I'd generally push for samples to not explicitly paginate unless it's the point of the sample.